.golangci.yaml

```diff
@@ -0,0 +1,37 @@
+run:
+  go: '1.17'
+  timeout: 30m
+  skip-files:
+    - "^zz_generated.*"
+
+issues:
+  max-same-issues: 0
+  # Excluding configuration per-path, per-linter, per-text and per-source
+  exclude-rules:
+    # exclude ineffassing linter for generated files for conversion
+    - path: conversion\.go
+      linters:
+        - ineffassign
+
+linters:
+  disable-all: true
+  enable: # please keep this alphabetized
+  # Don't use soon to deprecated[1] linters that lead to false
+  # https://github.com/golangci/golangci-lint/issues/1841
+  # - deadcode
+  # - structcheck
+  # - varcheck
+    - ineffassign
+    - staticcheck
+    - stylecheck
+    - unused
```

Comment:
Just as https://github.com/etcd-io/etcd/issues/14164 pointed out, some issues are successfully discovered by [revive](https://github.com/mgechev/revive), so it would be better if we can enable it. But of course, we can enhance the golangci.yaml step by step in separate PR(s).

Reply:
I fully agree, @ahrtr .

Adding `revive`'s and a subset of its linters will require careful consideration/discussion from all of us — as others have pointed in #14164, we have to be careful with potential etcd developer newcomers.